src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Runtimes/MissingSdkRuntimeDetector.cs

```diff
@@ -57,29 +67,26 @@ protected override async Task InitializeCoreAsync(CancellationToken cancellation
             // safe to use within IProjectDynamicLoadComponent.LoadAsync.
             _projectGuid = await _project.UnconfiguredProject.GetProjectGuidAsync();
             _missingSetupComponentRegistrationService.RegisterProjectConfiguration(_projectGuid, _project);
-            _projectFaultHandlerService.Forget(RegisterSdkNetCoreRuntimeNeededInProjectAsync(_project), _project.UnconfiguredProject);
+            _subscription = _projectSubscriptionService.ProjectRuleSource.SourceBlock.LinkToAsyncAction(target: OnProjectChangedAsync, _project.UnconfiguredProject);
         }
 
         private async Task RegisterSdkNetCoreRuntimeNeededInProjectAsync(ConfiguredProject project)
         {
-            if (IsInitialized)
-            {
-                var projectProperties = project.Services.ExportProvider.GetExportedValue<ProjectProperties>();
+            var projectProperties = project.Services.ExportProvider.GetExportedValue<ProjectProperties>();
 
-                ConfigurationGeneral configuration = await projectProperties.GetConfigurationGeneralPropertiesAsync();
+            ConfigurationGeneral configuration = await projectProperties.GetConfigurationGeneralPropertiesAsync();
 
-                string? targetFrameworkIdentifier = await configuration.TargetFrameworkIdentifier.GetDisplayValueAsync();
+            string? targetFrameworkIdentifier = await configuration.TargetFrameworkIdentifier.GetDisplayValueAsync();
 
-                string? targetFrameworkVersion = await configuration.TargetFrameworkVersion.GetDisplayValueAsync();
+            string? targetFrameworkVersion = await configuration.TargetFrameworkVersion.GetDisplayValueAsync();
 
-                if (!string.Equals(targetFrameworkIdentifier, s_netCoreTargetFrameworkIdentifier, StringComparisons.FrameworkIdentifiers) ||
-                    string.IsNullOrEmpty(targetFrameworkVersion))
-                {
-                    targetFrameworkVersion = string.Empty;
-                }
-                
-                _missingSetupComponentRegistrationService.RegisterPossibleMissingSdkRuntimeVersion(_projectGuid, project, targetFrameworkVersion);
+            if (!string.Equals(targetFrameworkIdentifier, s_netCoreTargetFrameworkIdentifier, StringComparisons.FrameworkIdentifiers) ||
+                string.IsNullOrEmpty(targetFrameworkVersion))
+            {
+                targetFrameworkVersion = string.Empty;
             }
+            
+            _missingSetupComponentRegistrationService.RegisterPossibleMissingSdkRuntimeVersion(_projectGuid, project, targetFrameworkVersion);
```

Comment:
Is it ok to call this method for every project update? That seems like a behaviour change to consider. I guess previously it was trying to call this only once.

For example, if I dismiss the popup, will it come back every time I make a change? Consider for example that you're making changes in the Project Properties UI. Every selection you make might make it re-appear.

Reply:
> will it come back every time I make a change

No, it won't. The gold bar will only appear once